src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java

```diff
@@ -232,6 +244,95 @@ void shouldUpdateTransactionToCheckIn() {
+
+    assertEquals(ITEM_ID, transactionItemHold.getFolioItemId());
+    assertEquals(INSTANCE_ID, transactionItemHold.getFolioInstanceId());
+    assertEquals(HOLDING_ID, transactionItemHold.getFolioHoldingId());
+    assertEquals(barcode, transactionItemHold.getFolioItemBarcode());
+    assertEquals(InnReachTransaction.TransactionState.TRANSFER, transaction.getState());
+    verify(innReachExternalService, times(1)).postInnReachApi(any(), any(), any());
+  }
+
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-another-inn-reach-transaction.sql",
+  })
+  void shouldSkipTransactionThatIsNotItemHold() {
+    var folioRequestId =  UUID.fromString("ea11eba7-3c0f-4d15-9cca-c8608cd6bc8a");
+    var event = getRequestDomainEvent(DomainEventType.UPDATED);
+    RequestDTO requestDTO = event.getData().getNewEntity();
+    event.setRecordId(null); // the listener should set this field from event key value
+    requestDTO.setId(folioRequestId);
+
+    var consumerRecord = new ConsumerRecord(CIRC_REQUEST_TOPIC, 1, 1, folioRequestId.toString(), event);
+    listener.handleRequestStorage(List.of(consumerRecord));
+    ArgumentCaptor<List<DomainEvent<RequestDTO>>> eventsCaptor = ArgumentCaptor.forClass(List.class);
+
+    verify(eventProcessor).process(eventsCaptor.capture(), any(Consumer.class));
+    var capturedEvents = eventsCaptor.getValue();
+    assertEquals(1, capturedEvents.size());
+
+    var capturedEvent = capturedEvents.get(0);
+    assertEquals(folioRequestId, capturedEvent.getRecordId());
+    var transaction = transactionRepository.fetchActiveByRequestId(requestDTO.getId()).orElse(null);
+    assertEquals(InnReachTransaction.TransactionState.PATRON_HOLD, transaction.getState());
+  }
+
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-another-inn-reach-transaction.sql"
+  })
+  void shouldSkipIfTransactionNotFoundByRequestId() {
+    var folioRequestId =  UUID.fromString("aa11eba7-3c0f-4d15-9cca-c8608cd6bc8a");
+    var event = getRequestDomainEvent(DomainEventType.UPDATED);
+
+    var consumerRecord = new ConsumerRecord(CIRC_REQUEST_TOPIC, 1, 1, folioRequestId.toString(), event);
+    listener.handleRequestStorage(List.of(consumerRecord));
+
+    ArgumentCaptor<List<DomainEvent<RequestDTO>>> eventsCaptor = ArgumentCaptor.forClass(List.class);
+    verify(eventProcessor).process(eventsCaptor.capture(), any(Consumer.class));
+    var capturedEvents = eventsCaptor.getValue();
+
+    var capturedEvent = capturedEvents.get(0);
+    assertEquals(folioRequestId, capturedEvent.getRecordId());
+    verify(transactionRepository, times(1)).fetchActiveByRequestId(any());
+    verify(itemService, times(0)).find(any());
+    verify(innReachExternalService, times(0)).postInnReachApi(any(), any(), any());
```

Comment:
just a tip, there is `verifyNoInteractions` for this purpose

Reply:
It is new to me. Thank you.